yellowbrick/features/pca.py

```diff
@@ -0,0 +1,176 @@
+    >>> X = iris.data
+    >>> y = iris.target
+    >>> pca_decomposition(X, color=y, proj_dim=3, colormap='RdBu_r')
+    
+    """
+    # Instantiate the visualizer
+    visualizer = PCADecomposition(X=X, y=y, ax=ax, scale=scale, proj_dim=proj_dim,
+                                  colormap= colormap, color=color)
+
+    # Fit and transform the visualizer (calls draw)
+    visualizer.fit(X, y, **kwargs)
+    visualizer.transform(X)
+
+    # Return the axes object on the visualizer
+    return visualizer.poof()
+##########################################################################
+##2D and #3D PCA Visualizer
+##########################################################################
+class PCADecomposition(DataVisualizer):
+    """
+    Produce a two or three dimensional principal component (PC) plot of a data set 
+    projected onto it first 2 or 3 PC. It is best practices to center and scale the inputted
+    data set before applying a PC decomposition. There are scale and center arguments
+    that can be used to control centering anc scaling of an inputted data set. Therefore 
+    this class is a one stop shop for easily getting a PC plot.
+
+    Parameters
+    ----------
+    X : ndarray or DataFrame of shape n x m
+        A matrix of n instances with m features
+        
+    y : ndarray or Series of length n
+        An array or series of target or class values
+
+    ax : matplotlib Axes, default: None
+        The axes to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+    
+    scale : bool, default: True
+        Boolean that indicates if user wants to scale data.
+    
+    proj_dim : int, default: 2
+        The dimension of the PCA project for visualizer.
+    
+    color : list or tuple of colors, default: None
+        Specify the colors for each individual class.
+
+    colormap : string or cmap, default: None
+        optional string or matplotlib cmap to colorize lines
+        Use either color to colorize the lines on a per class basis or
+        colormap to color them on a continuous scale.
+    
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+    >>> from sklearn import datasets
+    >>> iris = datasets.load_iris()
+    >>> X = iris.data
+    >>> y = iris.target
+    >>> params = {'scale': True, 'center': False, 'col': y}
+    >>> visualizer = PCADecomposition(**params)
+    >>> visualizer.fit(X)
+    >>> visualizer.transform(X)
+    >>> visualizer.poof()
+
+    """
+    def __init__(self, X=None, y=None, ax=None, scale=True, color=None, proj_dim=2,
+                 colormap=palettes.DEFAULT_SEQUENCE, **kwargs):
+        super(PCADecomposition, self).__init__(ax=ax, **kwargs)
+        # Data Parameters
+        if proj_dim not in (2, 3):
+            raise ValueError("proj_dim object is not 2 or 3.")
```

Comment:
It would be my preference that you raised a `YellowbrickValueError` (from the exceptions.py module) so that we can disambiguate between Yellowbrick exceptions and other exceptions. Note that `YellowbrickValueError` extends both `YellowbrickException` and `ValueError` so you can test for both as you're doing in the tests.